dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteServiceCollectionExtensions.cs

```diff
@@ -60,24 +43,9 @@ public static IServiceCollection AddSqliteVectorStore(
         string connectionString,
         SqliteVectorStoreOptions? options = default,
         string? serviceId = default)
-    {
-        services.AddKeyedTransient<IVectorStore>(
+        => services.AddKeyedSingleton<IVectorStore>(
```

Comment:
Note the change from transient to singleton lifecycle here - this aligns with other connectors.